04.class/app.js

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env node
+
+import minimist from "minimist";
+import readline from "readline";
+import enquirer from "enquirer";
+const { Select } = enquirer;
+import MemoDb from "./memo_db.js";
+
+class App {
+  constructor() {
+    this.memoDb = new MemoDb();
+  }
+
+  async execute() {
+    const args = minimist(process.argv.slice(2));
+
+    if (args.l) {
+      await this.#listMemos();
+    } else if (args.r) {
+      await this.#readMemo();
+    } else if (args.d) {
+      await this.#deleteMemo();
+    } else if (args._.length === 0) {
+      await this.#createMemo();
+    }
+  }
+
+  async #listMemos() {
+    const memos = await this.memoDb.loadMemos();
+    memos.forEach((memo, index) =>
+      console.log(`${index + 1}: ${memo.firstLine()}`),
+    );
+  }
+
+  async #readMemo() {
+    await this.#chooseMemoAction(
+      "Choose a note you want to see:",
+      async (id) => {
+        const memo = await this.memoDb.selectMemo(id);
+        if (memo) {
+          console.log(memo.content);
+        }
+      },
+    );
+  }
+
+  async #deleteMemo() {
+    await this.#chooseMemoAction(
+      "Choose a note you want to delete:",
+      async (id) => {
+        await this.memoDb.deleteMemo(id);
+      },
+    );
+  }
+
+  async #createMemo() {
+    console.log("Write a note:");
+    const memoContent = await this.#inputMemoContent();
+    await this.memoDb.insertMemo(memoContent.trim());
+  }
+
+  #inputMemoContent() {
+    return new Promise((resolve) => {
+      let allInput = "";
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+      });
+
+      rl.on("line", (input) => (allInput += input + "\n"));
+      rl.on("close", () => resolve(allInput));
+    });
+  }
+
+  async #chooseMemoAction(actionMessage, actionCallback) {
```

Comment:
```suggestion
  async #chooseAction(actionMessage, actionCallback) {
```

memoはなくしても大丈夫かも。
https://bootcamp.fjord.jp/pages/avoid-too-specific-var-name

同様に引数のactionもカットできそうです。

```suggestion
  async #chooseAction(message, callback) {
```

Reply:
`chooseMemoAction()`のメソッド名およびその引数名を修正しました。